### ISSUE TITLE
ToTaToNet Pyannote Bug Report

### ISSUE THREAD

**Issue by ruixCMU** [joonaskalda/PixIT#4]
When we do not finetune WavLM in the Pyannote implementation of ToTaToNet, the model does not update its weight due to that they set automatic_optimization = False in the constructor of the model, which disabled optimizer update. They also did not do any manual optimization for the case of not finetuning the WavLM in the PixIT task either.

You could fix this by enabling automatic_optimization, hope this helps.

**Comment by ruixCMU** [comment]
Also, when finetune_wavlm is set to true, it seems that the wavlm is still in eval mode, so we need to manually set it to train.

**Comment by joonaskalda** [comment]
Thank you for investigating these issues and sorry for the delay in responding! 

The first issue has now been fixed with [this pyannote.audio PR](https://github.com/pyannote/pyannote-audio/pull/1791 ).

Not setting WavLM to train mode (which enables dropout and layer normalization) was at first unintentional but I looked into it and interestingly enabling it actually degrades the performance on AMI-SDM.

**Comment by ruixCMU** [comment]
Thank you for your response. I noticed that they fixed the first issue, which was also reported by me.

It's interesting to know that disabling the Dropout and Normalization gives better performance. Maybe it indicates underfitting.
